dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/DefaultQueryItemLocator.java

```diff
@@ -105,8 +103,26 @@ public QueryItem getQueryItemFromDimension( String dimension, Program program, E
         return getDataElement( dimension, program, legendSet, type )
             .orElseGet( () -> getTrackedEntityAttribute( dimension, program, legendSet )
                 .orElseGet( () -> getProgramIndicator( dimension, program, legendSet )
-                    .orElseThrow(
-                        () -> new IllegalQueryException( new ErrorMessage( ErrorCode.E7224, dimension ) ) ) ) );
+                    .orElseGet( () -> getDynamicDimension( dimension )
+                        .orElseThrow(
+                            () -> new IllegalQueryException( new ErrorMessage( ErrorCode.E7224, dimension ) ) ) ) ) );
+    }
+
+    private Optional<QueryItem> getDynamicDimension( String dimension )
```

Comment:
Maybe just add some Javadoc here to explain the goal of this method and its limitations (if any).